bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/discovery/JellyfinClientDiscoveryService.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.jellyfin.internal.discovery;
+
+import static org.openhab.binding.jellyfin.internal.JellyfinBindingConstants.DISCOVERY_RESULT_TTL_SEC;
+import static org.openhab.binding.jellyfin.internal.JellyfinBindingConstants.THING_TYPE_CLIENT;
+import static org.openhab.binding.jellyfin.internal.JellyfinBindingConstants.THING_TYPE_SERVER;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.jellyfin.sdk.api.client.exception.ApiClientException;
+import org.jellyfin.sdk.api.client.exception.InvalidStatusException;
+import org.jellyfin.sdk.model.api.SessionInfo;
+import org.openhab.binding.jellyfin.internal.handler.JellyfinServerHandler;
+import org.openhab.binding.jellyfin.internal.util.SyncCallback;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link JellyfinClientDiscoveryService} discover Jellyfin clients connected to the server.
+ *
+ * @author Miguel Alvarez - Initial contribution
+ */
+@NonNullByDefault
+public class JellyfinClientDiscoveryService extends AbstractDiscoveryService {
+    private final Logger logger = LoggerFactory.getLogger(JellyfinClientDiscoveryService.class);
+    private final JellyfinServerHandler serverHandler;
+
+    public JellyfinClientDiscoveryService(JellyfinServerHandler serverHandler) throws IllegalArgumentException {
+        super(Set.of(THING_TYPE_SERVER), 60);
+        this.serverHandler = serverHandler;
+    }
+
+    @Override
+    protected void startScan() {
+        if (!serverHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+            logger.warn("Server handler {} is not online.", serverHandler.getThing().getLabel());
+            return;
+        }
+        logger.info("Searching devices for server {}", serverHandler.getThing().getLabel());
```

Comment:
Logging to info should be used sparsely e.g. a newly started component or a user file that has been loaded. This could be debug. See https://next.openhab.org/docs/developer/guidelines.html#f-logging